### PR TITLE
add support for password authentication against Domain Display Name

### PR DESCRIPTION
**Pull request by surendarchandra**
# Description

In 5.0, the optional Domain Display Name allows one to specify the authentication domain. If unspecified, Rubrik CDM will first the "local" domain before searching through all the configured LDAP domains. This could be slow and create many false authentication failures. Hence, add an option to Connect() to specify the Domain Display Name.

This change then switches from password authentication to a token based authentication; potentially speeding up further interactions.

## Related Issue

https://github.com/rubrikinc/rubrik-sdk-for-python/issues/104

## Motivation and Context

For customers who have many LDAP integrations, the current behavior will lead to many
false alarms.

## How Has This Been Tested?

Tested against an internal Brik.

_$ python_
Python 3.7.3 (default, Mar 27 2019, 09:23:15)
[Clang 10.0.1 (clang-1001.0.46.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> _import rubrik_cdm_
>>> _rubrik=rubrik_cdm.Connect(username='surendar@rubrik-lab.com', domain_display_name='Rubrik Lab')_
/Users/surendar/rubrik-sdk-for-python/venv/lib/python3.7/site-packages/urllib3-1.25.3-py3.7.egg/urllib3/connectionpool.py:851: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
>>> _rubrik.cluster_version()_
/Users/surendar/rubrik-sdk-for-python/venv/lib/python3.7/site-packages/urllib3-1.25.3-py3.7.egg/urllib3/connectionpool.py:851: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
'5.0.2-1614'

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **[CONTRIBUTION](https://github.com/rubrikinc/rubrik-sdk-for-python/blob/master/CONTRIBUTING.md)** document.
- [ ] I have updated the CHANGELOG file accordingly for the version that this merge modifies.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by drew-russell** [comment]
@surendarchandra can you switch this to the `devel` branch? All new code should land their first before hitting `master`, which should match the code available in the latest release only.

File: rubrik_cdm/rubrik_cdm.py [hunk 1]
```diff
@@ -112,6 +112,19 @@ def __init__(self, node_ip=None, username=None, password=None, api_token=None, e
                     self.password = password
                     self.log("Password: *******\n")
 
+                # Domain display name may be specified to force authentication to this domain
+                session_url = 'https://{}/api/internal/session'.format(self.node_ip)
+                ddn = domain_display_name if domain_display_name else os.environ.get('rubrik_cdm_domain')
```
Comment (drew-russell) [inline rubrik_cdm/rubrik_cdm.py]: Can you break this out into something more verbose? Probably contrary to what you're used to but one of the main tenants of the SDK is having the code be as readable as possible to someone who is just getting started with programming and one-liners can be confusing for them

File: rubrik_cdm/rubrik_cdm.py [hunk 2]
```diff
@@ -112,6 +112,19 @@ def __init__(self, node_ip=None, username=None, password=None, api_token=None, e
...
+                if ddn:
+                    session_url = session_url + '/realm/{}'.format(ddn)
+
+                # Get the web API session token to make future interactions faster
+                response = requests.post(session_url, data='{"initParams": {}}', auth=(self.username, self.password), verify=False)
```
Comment (drew-russell) [inline rubrik_cdm/rubrik_cdm.py]: What happens if this call fails? Connectivity issues etc.

We also need to have a timeout value of some sort here. Default throughout the SDK is 15 seconds but it's not unheard-of for customers to need to increase that.

**Comment by drew-russell** [comment]
@surendarchandra we pushed changes to the logic here which is going to affect the placement of this code. Can you rebase/merge and update as needed as well? Apologizes on that one.

**Comment by surendarchandra** [comment]
Yeah, saw that the branch had changed. Tied up in other stuff; working on rebasing to the proper code.

**Pull request closed without merging**